ui/i18n.py

```diff
@@ -0,0 +1,24 @@
+import gettext
+
+
+class I18n:
+    def __init__(self, locale='en_GB:en'):
+        if locale:
+            self.lang = locale
+        else:
+            self.lang = None
+        self.translate = None
+
+    def _(self, key):
+        if self.lang is not None:
+            try:
+                language = self.lang
+                translate = gettext.translation(
+                    'canute', localedir='ui/locale', languages=[language], fallback=False)
+                translate.install()
+            except Exception:
+                translate = gettext.NullTranslations()
+            self.translate = translate
+        elif not self.translate:
+            self.translate = gettext.NullTranslations()
+        return self.translate.gettext(key)
```

Comment:
Can you explain the logic behind the `I18n._`  method? It seems a tad convoluted. Things under the first `if` seem like they should be in `__init__`. 

Not a big fan of having a `class` with a single method either. We can probably refactor that to just be a closure.

Reply:
This has changed quite a bit since you commented here.  So, the gettext call does the parsing to the POT file but also does the translate.  It needs to be dynamic since we want to be able to update the language at runtime.  We store the translated text in the instance, at the moment, but we don't currently need to.

Reply:
My comment is attached to a specific diff so unless you haven't pushed changes I have commented on an up to date version. The comment hides itself once it goes out of date. 

Anyway, the way this is done needs to be cleaned up. Maybe the [class based API](https://docs.python.org/3/library/gettext.html#class-based-api) is what you want? 

> if your application needs to switch languages on the fly, you probably want to use the class-based API instead.